### PR TITLE
CMake install targets and a small fix for the interface target

### LINKED ISSUE #41: test_package to verify installed cmake macros
> > Thanks @duca! I pushed some minor changes. For testing, I just installed it to a subdirectory and inspected the files. How do you test it?
> > Does it make sense to add a package test similar how Conan does it? https://github.com/conan-io/conan-center-index/tree/master/recipes/open62541/all/test_package
> 
> I think it does make sense, I built a stub project and optimized those lines and did some minor changes to make it work properly

@lukasberbuer any suggestions on how to integrate the test_package with the test fixture?

### PR DESCRIPTION
**Pull request by duca**
In this PR, I fixed the usage of the open62541pp_project_options target for modern cmake and added the installation of the infrastructure necessary to find_package this library when `lib/cmake/open62541pp` path is appended to the user's CMAKE_MODULE_PATH list

resolves #41

### CODE REVIEW

File: cmake/open62541ppConfig.cmake.in [hunk 1]
```diff
@@ -0,0 +1,17 @@
+@PACKAGE_INIT@
+
+include("${CMAKE_CURRENT_LIST_DIR}/open62541ppTargets.cmake")
+
+set (open62541pp_TOOLS_DIR @PACKAGE_open62541pp_install_tools_dir@ CACHE PATH "Path to the directory that contains the tooling of the stack")
+set (open62541pp_NODESET_DIR @PACKAGE_open62541pp_install_nodeset_dir@ CACHE PATH "Path to the directory that contains the OPC UA schema repository")
```
Comment (lukasberbuer) [inline cmake/open62541ppConfig.cmake.in:6]: Do we need this? The open62541 tooling and OPC UA schema should be part of the open62541 package and not ours?

Comment (duca) [reply cmake/open62541ppConfig.cmake.in:6]: Lines 5 and 6 were not created by me, maybe I forgot to remove them when I rebased against your master branch?

**Comment by lukasberbuer** [comment]
Hi @duca, I pushed some minor changes. For testing, I just installed it to a subdirectory and inspected the files.
How do you test it?

Does it make sense to add a package test similar how Conan does it? https://github.com/conan-io/conan-center-index/tree/master/recipes/open62541/all/test_package

**Comment by duca** [comment]
> Thanks @duca! I pushed some minor changes. For testing, I just installed it to a subdirectory and inspected the files. How do you test it?
> 
> Does it make sense to add a package test similar how Conan does it? https://github.com/conan-io/conan-center-index/tree/master/recipes/open62541/all/test_package

I think it does make sense, I built a stub project and optimized those lines and did some minor changes to make it work properly

File: cmake/open62541ppConfig.cmake.in (hunk 1 above)
Comment (duca) [reply cmake/open62541ppConfig.cmake.in]: No we didn't, this is a leftover from the copy I did of the original open62541 template file

**Comment by duca** [comment]
> > Thanks @duca! I pushed some minor changes. For testing, I just installed it to a subdirectory and inspected the files. How do you test it?
> > Does it make sense to add a package test similar how Conan does it? https://github.com/conan-io/conan-center-index/tree/master/recipes/open62541/all/test_package
> 
> I think it does make sense, I built a stub project and optimized those lines and did some minor changes to make it work properly

@lukasberbuer any suggestions on how to integrate the test_package with the test fixture?

**Comment by lukasberbuer** [comment]
Hi @duca, I think something went wrong with the last two commits. Let's revert them and merge this PR.
After reading [this](https://cmake.org/cmake/help/v3.22/guide/using-dependencies/index.html), I'm quite confident, that we have a good solution here. Let's tackle the package test (#41) in another PR.

**Comment by ErikTempelaarVO** [comment]
Nice work, with these latest changes a simple 'cmake.install()' from a conan-recipe does the trick.

**Pull request merged**